test/test_decoders.py

```diff
@@ -126,6 +126,27 @@ def test_metadata(self, seek_mode):
         assert decoder.metadata.height == 270
         assert decoder.metadata.width == 480
 
+    def test_create_bytes_ownership(self):
+        # Note that the bytes object we use to instantiate the decoder does not
+        # live past the VideoDecoder destructor. That is what we're testing:
+        # that the VideoDecoder takes ownership of the bytes. If it does not,
+        # then we will hit errors when we try to actually decode from the bytes
+        # later on. By the time we actually decode, the reference on the Python
+        # side has gone away, and if we don't have ownership on the C++ side, we
+        # will hit runtime errors or segfaults.
+        #
+        # Also note that if this test fails, OTHER tests will likely
+        # mysteriously fail. That's because a failure in this tests likely
+        # indicates memory corruption, and the memory we corrupt could easily
+        # cause problems in other tests. So if this test fails, fix this test
+        # first.
+        with open(NASA_VIDEO.path, "rb") as f:
+            decoder = VideoDecoder(f.read())
```

Comment:
I think we should call `gc.collect()` here, to force the GC to run and free the resources? It won't change anything in this branch, but in `main`, I think this would deterministically ensure that the resource gets freed. I may be wrong but I suspect there might be cases where this test *doesn't* segfault on `main` simply because the GC didn't collect things yet.